index.d.ts

```diff
@@ -0,0 +1,97 @@
+import { FastifyPluginAsync } from "fastify";
+import { MercuriusPlugin } from "mercurius";
+export interface PolicyFieldOptions {
+  ttl?: number;
+  storage?: MercuriusCacheStorage;
+  extendKey?: Function;
+  skip?: Function;
+  invalidate?: Function;
+  references?: Function;
+}
+
+export type PolicyFieldName = string;
+export type PolicyField = Record<PolicyFieldName, PolicyFieldOptions | object>;
+export type PolicyName = string;
+export type MercuriusCachePolicy = Record<PolicyName, PolicyField>;
+
+export interface MercuriusCacheStorageOptions {
```

Comment:
MercuriusCacheStorageOptions can be either MercuriusCacheStorageMemoryOptions or MercuriusCacheStorageRedisOptions

they are (sorry for jsdoc format)

```js

/**
 * @typedef StorageMemoryOptions
 * @property {?number} [size=1024]
 * @property {?Logger} [log]
 * @property {?boolean} [invalidation=false]
 */

/**
 * @typedef StorageRedisOptions
 * @property {!object} client
 * @property {?Logger} log
 * @property {?Object|boolean} [invalidation=false]
 * @property {?number} [invalidation.referencesTTL=60]
 */

```